source/common/buffer/watermark_buffer.cc

```diff
@@ -210,18 +210,27 @@ uint64_t WatermarkBufferFactory::resetAccountsGivenPressure(float pressure) {
       std::floor(pressure * BufferMemoryAccountImpl::NUM_MEMORY_CLASSES_) + 1, 8);
 
   uint32_t last_bucket_to_clear = BufferMemoryAccountImpl::NUM_MEMORY_CLASSES_ - buckets_to_clear;
-  ENVOY_LOG_MISC(warn, "resetting streams in buckets >= {}", last_bucket_to_clear);
+  bool need_message = false;
 
   // Clear buckets, prioritizing the buckets with larger streams.
   uint32_t num_streams_reset = 0;
   for (uint32_t buckets_cleared = 0; buckets_cleared < buckets_to_clear; ++buckets_cleared) {
     const uint32_t bucket_to_clear =
         BufferMemoryAccountImpl::NUM_MEMORY_CLASSES_ - buckets_cleared - 1;
-    ENVOY_LOG_MISC(warn, "resetting {} streams in bucket {}.",
-                   size_class_account_sets_[bucket_to_clear].size(), bucket_to_clear);
+    absl::flat_hash_set<BufferMemoryAccountSharedPtr>& bucket =
+        size_class_account_sets_[bucket_to_clear];
+    if (bucket.empty()) {
+      // Don't print a message if there is nothing to clear.
+      continue;
+    }
+    if (need_message) {
+      need_message = false;
```

Comment:
I think what you're trying to do here is print this message one per call if we have a bucket with some entity in it.

Consider inverting the logic e.g. 
```
bool printed_message = false; 
... // skip empty buckets
if(!printed_message) {
 printed_message = true;
 warn(...)
}
```

You could also consider having a log at the end like `ENVOY_LOG_EVERY_NTH` for if the buckets were empty every say 200 calls.

Reply:
LOG_EVERY_NTH is a good idea; done.

I don't see why the inverted logic is better.